src/main/java/com/alipay/oceanbase/hbase/OHTable.java

```diff
@@ -1677,36 +1631,73 @@ private KeyValue modifyQualifier(KeyValue original, byte[] newQualifier) {
         BatchOperation batch = obTableClient.batchOperation(tableName);
-
-        for (Map.Entry<byte[], List<KeyValue>> entry : familyMap.entrySet()) {
-            byte[] family = entry.getKey();
-            List<KeyValue> keyValueList = entry.getValue();
-            for (KeyValue kv : keyValueList) {
-                if (qualifiers != null) {
-                    qualifiers.add(kv.getQualifier());
+        if (actions != null) {
+            int posInList = -1;
+            int singleOpResultNum;
+            for (Row row : actions) {
+                singleOpResultNum = 0;
+                posInList++;
+                if (!(row instanceof Put) && !(row instanceof Delete)) {
+                    throw new FeatureNotSupportedException(
+                        "not supported other type in batch yet,only support put and delete");
+                } else if (row instanceof Put) {
+                    Put put = (Put) row;
+                    if (put.isEmpty()) {
+                        throw new IllegalArgumentException("No columns to insert for #"
+                                                           + (posInList + 1) + " item");
+                    }
+                    for (Map.Entry<byte[], List<KeyValue>> entry : put.getFamilyMap().entrySet()) {
+                        byte[] family = entry.getKey();
+                        List<KeyValue> keyValueList = entry.getValue();
+                        for (KeyValue kv : keyValueList) {
+                            singleOpResultNum++;
+                            if(isTableGroup){
+                                KeyValue new_kv = modifyQualifier(kv,(Bytes.toString(family) + "." + Bytes.toString(kv.getQualifier()))
+                                    .getBytes());
+                                batch.addOperation(buildMutation(new_kv, INSERT_OR_UPDATE, isTableGroup));
+                            } else {
+                                batch.addOperation(buildMutation(kv, INSERT_OR_UPDATE, isTableGroup));
+                            }
+                        }
+                    }
+                } else {
+                    Delete delete = (Delete) row;
+                    if (delete.isEmpty()) {
+                        singleOpResultNum++;
+                        KeyValue kv = new KeyValue(delete.getRow(), delete.getTimeStamp(),
+                            KeyValue.Type.Maximum);
+                        batch.addOperation(buildMutation(kv, DEL, isTableGroup));
+                    } else {
+                        for (Map.Entry<byte[], List<KeyValue>> entry : delete.getFamilyMap()
+                            .entrySet()) {
+                            byte[] family = entry.getKey();
+                            List<KeyValue> keyValueList = entry.getValue();
+                            List<com.alipay.oceanbase.rpc.mutation.Mutation> mutations = new LinkedList<>();
```

Comment:
why not use import `com.alipay.oceanbase.rpc.mutation.Mutation` and use Mutation directly